sdks/java/io/bigquery-io-perf-tests/src/test/java/org/apache/beam/sdk/bigqueryioperftests/BigQueryIOReadPerformanceIT.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.beam.sdk.testutils.metrics.IOITMetrics;
+import org.apache.beam.sdk.testutils.metrics.MetricsReader;
+import org.apache.beam.sdk.testutils.metrics.TimeMonitor;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * A performance test of {@link org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO}.
+ *
+ * <p>Usage:
+ *
+ * <pre>
+ *  ./gradlew integrationTest -p sdks/java/io/gcp/bigquery -DintegrationTestPipelineOptions='[
+ *  "--bigQuerySourceDataset=source-dataset",
+ *  "--bigQuerySourceTable=source-table",
+ *  "--bigQueryDataset=metrics-dataset",
+ *  "--bigQueryTable=metrics-table",
+ *  "--sourceOptions={"numRecords":"1000", "keySize":1, valueSize:"1024"}
+ *  }"]'
+ *  --tests org.apache.beam.sdk.io.gcp.bigQuery.BigQueryIOReadPerformanceIT
+ *  -DintegrationTestRunner=direct
+ * </pre>
+ */
+@RunWith(JUnit4.class)
+public class BigQueryIOReadPerformanceIT {
+  private static final String NAMESPACE = BigQueryIOReadPerformanceIT.class.getName();
+  private static String bigQueryMetricsTable;
+  private static String bigQueryMetricsDataset;
+  private static String bigQuerySourceDataset;
+  private static String bigQuerySourceTable;
+  private static SyntheticSourceOptions sourceOptions;
+  private static String tableQualifier;
+  private static String tempRoot;
+  private static BigQueryPerfTestOptions options;
+
+  /** Options for this io performance test. */
+  public interface BigQueryPerfTestOptions extends IOTestPipelineOptions {
+    @Description("Synthetic source options")
+    @Validation.Required
+    String getSourceOptions();
+
+    void setSourceOptions(String value);
+
+    @Description("BQ dataset with source data")
+    String getBigQuerySourceDataset();
+
+    void setBigQuerySourceDataset(String dataset);
+
+    @Description("BQ table with source data")
+    String getBigQuerySourceTable();
+
+    void setBigQuerySourceTable(String table);
+  }
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    options = IOITHelper.readIOTestPipelineOptions(BigQueryPerfTestOptions.class);
+    tempRoot = options.getTempRoot();
+    sourceOptions =
+        SyntheticOptions.fromJsonString(options.getSourceOptions(), SyntheticSourceOptions.class);
+    bigQueryMetricsDataset = options.getBigQueryDataset();
+    bigQueryMetricsTable = options.getBigQueryTable();
+    bigQuerySourceDataset = options.getBigQuerySourceDataset();
+    bigQuerySourceTable = options.getBigQuerySourceTable();
+    BigQueryOptions bigQueryOptions = BigQueryOptions.newBuilder().build();
+    tableQualifier =
+        String.format(
+            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    if (!checkForSourceTable()) {
+      createAndFillSourceTable();
+    }
+  }
```

Comment:
Do we need cleanup here? I think we should delete the BQ table after the test at least. Is there something else?

Reply:
You're right, I'll add the cleanup